libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigator.kt

```diff
@@ -23,51 +26,222 @@ interface MapboxNativeNavigator {
+     */
     fun getStatus(date: Date): TripStatus
 
     // Routing
 
+    /**
+     * Sets the route path for the navigator to process.
+     * Returns initialized route state if no errors occurred.
+     * Otherwise, it returns a invalid route state.
+     *
+     * @param route [DirectionsRoute] to follow.
+     * @param routeIndex Which route to follow
+     * @param legIndex Which leg to follow
+     *
+     * @return a [NavigationStatus] route state if no errors occurred.
+     * Otherwise, it returns a invalid route state.
+     */
     fun setRoute(
         route: DirectionsRoute,
         routeIndex: Int = INDEX_FIRST_ROUTE,
         legIndex: Int = INDEX_FIRST_LEG
     ): NavigationStatus
 
+    /**
+     * Updates annotations so that subsequent calls to getStatus will
+     * reflect the most current annotations for the route
+     *
+     * @param legAnnotationJson A string containing the json/pbf annotations
+     * @param routeIndex Which route to apply the annotation update to
+     * @param legIndex Which leg to apply the annotation update to
+     *
+     * @return True if the annotations could be updated false if not (wrong number of annotations)
+     */
     fun updateAnnotations(legAnnotationJson: String, routeIndex: Int, legIndex: Int): Boolean
+
+    /**
+     * Gets the banner at a specific step index in the route. If there is no
+     * banner at the specified index, we will do something.
+     *
+     * @param index Which step you want to get [BannerInstruction] for
+     *
+     * @return [BannerInstruction] for step index you passed
+     */
     fun getBannerInstruction(index: Int): BannerInstruction?
+
+    /**
+     * Gets a polygon around the currently loaded route. The method uses a bitmap approach
+     * in which you specify a grid size (pixel size) and a dilation (how many pixels) to
+     * expand the intial grid cells that are interesected by the route.
+     *
+     * @param gridSize the size of the individual grid cells
+     * @param bufferDilation the number of pixels to dilate the initial intersection by it can
+     * be thought of as controlling the halo thickness around the route
+     *
+     * @return a geojson as [String] representing the route buffer polygon
+     */
     fun getRouteGeometryWithBuffer(
         gridSize: Float = GRID_SIZE,
         bufferDilation: Short = BUFFER_DILATION
     ): String?
 
+    /**
+     * Follows a new route and leg of the already loaded directions
+     * Returns an initialized route state if no errors occurred
+     * otherwise, it returns an invalid route state.
+     *
+     * @param routeIndex new route index
+     * @param legIndex new leg index
+     *
+     * @return an initialized route state as [NavigationStatus]
+     */
     fun updateLegIndex(routeIndex: Int, legIndex: Int): NavigationStatus
 
     // Free Drive
 
+    /**
+     * Uses valhalla and local tile data to generate electronic horizon json
```

Comment:
may be use smth abstract like `Routing Engine` instead of `valhalla`.  Here and below

Reply:
👍